### PR TITLE
Update Referrers API to match RC.1

### LINKED ISSUE #188: Update Referrers API to match RC.1
To achieve max compatibility, we need to try the referrer API defined by `v1.0.0-rc.1`. Then fallback to `v1.0.0-draft.1` if failed.

Reference: https://github.com/oras-project/artifacts-spec/blob/v1.0.0-rc.1/manifest-referrers-api.md

### PR DESCRIPTION
**Pull request by m5i-work**
Resolves #188

### CODE REVIEW

**Comment by northtyphoon** [comment]
@m5i-work just want to make sure you are aware of, there is a change in rc1 spec regarding to pagination. The link url has been changes to relative url. https://github.com/oras-project/artifacts-spec/blob/v1.0.0-rc.1/manifest-referrers-api.md#paging-results
Not sure if existing code can handle both. You may consider to incorporate in current or future pr.

File: registry/remote/url.go [hunk 1]
```diff
@@ -99,3 +98,16 @@ func buildArtifactReferrerURL(plainHTTP bool, ref registry.Reference) string {
 		ref.Reference,
 	)
 }
+
+// buildArtifactReferrerURL builds the URL for accessing the manifest referrers API in artifact spec v1.0.0-rc.1.
+// Format: <scheme>://<registry>/v2/<repository>/_oras/artifacts/referrers?digest=<digest>
+// Reference: https://github.com/oras-project/artifacts-spec/blob/v1.0.0-rc.1/manifest-referrers-api.md
+func buildArtifactReferrerURL(plainHTTP bool, ref registry.Reference) string {
```
Comment (Wwwsylvia) [inline registry/remote/url.go:105]: Have we tested this in a real remote server?

File: registry/remote/url.go [hunk 2]
```diff
@@ -99,3 +98,16 @@ func buildArtifactReferrerURL(plainHTTP bool, ref registry.Reference) string {
...
+	return fmt.Sprintf(
+		"%s://%s/v2/%s/_oras/artifacts/referrers?digest=%s",
+		buildScheme(plainHTTP),
+		ref.Host(),
+		ref.Repository,
```
Comment (shizhMSFT) [inline registry/remote/url.go:110]: The referrer API in the latest spec falls in the `/v2/` path. Should we leverage `buildRepositoryBaseURL` in `buildArtifactReferrerURL`?

**Comment by shizhMSFT** [comment]
Note: Since this PR is for rc, let's merge it after #195

File: registry/remote/url.go (hunk 1 above)
Comment (Wwwsylvia) [reply registry/remote/url.go:105]: Great!

File: registry/remote/repository_test.go [hunk 3]
```diff
@@ -875,11 +875,16 @@ func TestRepository_Predecessors(t *testing.T) {
...
+				t.Errorf("digest not provided or mismatch: %s %q", r.Method, r.URL)
+				w.WriteHeader(http.StatusBadRequest)
+				return
+			}
 			referrers = referrerSet[0]
 			w.Header().Set("Link", fmt.Sprintf(`<%s?n=2&test=foo>; rel="next"`, path))
 		}
 		result := struct {
-			References []artifactspec.Descriptor `json:"references"`
+			References []artifactspec.Descriptor `json:"referrers"`
```
Comment (Wwwsylvia) [inline registry/remote/repository_test.go:887]: `Referrers []artifactspec.Descriptor` ?

File: registry/remote/repository.go [hunk 4]
```diff
@@ -376,18 +386,28 @@ func (r *Repository) referrers(ctx context.Context, desc ocispec.Descriptor, fn
 	}
 	defer resp.Body.Close()
 
+	if resp.StatusCode == http.StatusNotFound {
+		return "", errdef.ErrNotFound
```
[the remark this answers is not in the document]
Comment (Wwwsylvia) [reply registry/remote/repository.go:390]: Yeah I mean if the `errdef.ErrorNotFound` is returned to users, it seems missing errors details like https://github.com/oras-project/oras-go/blob/main/registry/remote/internal/errutil/errors.go#L83.

**Review by Wwwsylvia — APPROVED** [review]
LGTM

**Comment by shizhMSFT** [comment]
> There are 8 commits in this PR. It is recommended to rebase and squash them into one.

It's okay since oras only allows "squash and merge" so that the main branch is always clean.

**Pull request merged**